### PR TITLE
Set container's network interface MAC address using its name hash

### PR DESCRIPTION
**Pull request by raneq**
What I'm doing here:

1. take container's name
2. hash it
3. take last 6 bytes to build a pseudo-random, static mac address
4. add it to the container settings

What are the expected consequences:

1. lxc respects this setting, instead of generating random mac addr
2. dnsmasq follow its default behavior of setting a static IP based on a pseudo-random function on mac addr [**1**].
3. each time a container is created with the same name, it will get the same IP addr assigned

This is tested to work in a Debian 9 and lxc version 3.0.3

---

**1**: dnsmasq alternative option where it explains also the default (and our desired) behavior:

>        --dhcp-sequential-ip
>               Dnsmasq  is designed to choose IP addresses for DHCP clients using a
>               hash of the client's MAC address. This normally  allows  a  client's
>               address  to  remain  stable long-term, even if the client  sometimes
>               allows its DHCP lease to expire. In this default mode  IP  addresses
>               are  distributed  pseudo-randomly  over the entire available address
>               range. There are sometimes circumstances (typically  server  deploy‐
>               ment) where it is more convenient to have IP addresses allocated se‐
>               quentially, starting from the lowest available address, and  setting
>               this  flag  enables  this  mode.  Note  that in the sequential mode,
>               clients which allow a lease to expire are much more likely  to  move
>               IP address; for this reason it should not be generally used.

### CODE REVIEW

File: config [hunk 1]
```diff
@@ -14,10 +14,34 @@ DEVENV_USER=${DEVENV_USER:-root}
...
+# > 50d858e0985ecc7f60418aaf0cc5ab587f42c2570a884095a9e8ccacd0f6545c  -
+# cut takes only the hash, leaving out the space and \n, but adds another \n
+# > 50d858e0985ecc7f60418aaf0cc5ab587f42c2570a884095a9e8ccacd0f6545c
+# tail takes the last 6 characters plus the \n, hence 7
+# > f6545c
+# sed captures groups of 2 chars 3 times and splits them with ':'
+# > :f6:54:5c
+function name_based_mac_addr() {
+  INPUT="$1"
+  echo -n "00:16:3e"
```
Comment (danypr92) [inline config:30]:
```suggestion
```

We don't need this `echo`, no?

**Review by danypr92 — CHANGES_REQUESTED** [review]
Great job @raneq!

File: config (hunk 1 above)
Comment (raneq) [reply config:30]: Yes we need it! It's half of the MAC address, its first 3 Bytes. It's a fixed prefix:

>     # Generate a MAC address using the 6 byte Xen prefix reserved for vm's
>     # plus 6 bytes generated by the hash of the container's name

However, I'm wrong with the comment: it's 3 Bytes + 3 Bytes, not 3 +6 :sweat_smile:

**Comment by enricostano** [comment]
@raneq nice! Would this work also with `systemd-networkd` DHCP service?

**Review by danypr92 — APPROVED** [review]

**Comment by danypr92** [comment]
> @raneq nice! Would this work also with `systemd-networkd` DHCP service?

Yes! I test it with the `systemd-networkd` and it's working!

**Comment by sauloperez** [comment]
What a professional @raneq ✊

**Pull request merged**